lib/trino-parquet/src/test/java/io/trino/parquet/reader/TestColumnReader.java

```diff
@@ -284,18 +290,28 @@ private static int pagesRowCount(List<RowRange> pageRowRanges)
     }
 
     /**
-     * @return Full cartesian product of arguments, i.e cartesianProduct({A, B}, {1,2}) will produce {{A,1}, {A,2}, {B,1}, {B,2}}
+     * @return Full cartesian product of arguments, i.e. cartesianProduct({{A, B}}, {{1,2}}) will produce {{A,1}, {A,2}, {B,1}, {B,2}}
      */
-    private static Object[][] cartesianProduct(Object[]... args)
+    private static Object[][] cartesianProduct(Object[][]... args)
     {
         return Lists.cartesianProduct(Arrays.stream(args)
                         .map(ImmutableList::copyOf)
                         .collect(toImmutableList()))
                 .stream()
-                .map(list -> list.toArray(Object[]::new))
+                .map(list -> flatten(list.toArray(Object[]::new)))
                 .toArray(Object[][]::new);
     }
 
+    /**
+     * If any of the arguments is an Object array itself, it is flattened
+     */
+    private static Object[] flatten(Object[] args)
+    {
+        return Arrays.stream(args)
+                .flatMap(arg -> arg instanceof Object[] ? Arrays.stream((Object[]) arg) : Stream.of(arg))
```

Comment:
in `cartesianProduct` you know array dimensions (`Object[][]`), so `instanceof` flexibility (and mental overhead) are not needed.

see above, remove `flatten`